tests/5.0/parallel_master_taskloop/test_parallel_master_taskloop.F90

```diff
@@ -0,0 +1,58 @@
+!/===--- test_parallel_master_taskloop.F90 -------------------------------------===//
+!
+! OpenMP API Version 5.0 Nov 2018
+!
+! This test checks the parallel master taskloop directive. The test performs
+! simple operations on an int array which are then checked for correctness.
+!
+!/===----------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+#define N 1024
+
+PROGRAM test_parallel_master_taskloop
+  USE iso_fortran_env
+  USE ompvv_lib
+  USE omp_lib
+  implicit none
+
+  OMPVV_TEST_VERBOSE(parallel_master_taskloop() .ne. 0)
+
+  OMPVV_REPORT_AND_RETURN()
+
+CONTAINS
+  INTEGER FUNCTION parallel_master_taskloop()
+    INTEGER:: errors = 0
+    INTEGER:: num_threads = -1
+    INTEGER, DIMENSION(N):: x, y, z
+    INTEGER:: i
+
+    OMPVV_INFOMSG("test_parallel_master_taskloop")
+
+    DO i = 1, N
+       x(i) = 1
+       y(i) = i + 1
+       z(i) = 2*(i + 1)
+    END DO
+
+    !$omp parallel master taskloop num_threads(OMPVV_NUM_THREADS_HOST) shared(x, y, z, num_threads) 
+    DO i = 1, N
+       x(i) = x(i) + y(i) * z(i)
+    END DO
+    IF (i .eq. 1) THEN
```

Comment:
Since IF is outside the loop it will never be executed.

Reply:
IMO since master is the only active thread in this context, it is okay to not guard this statement.

Reply:
> Since IF is outside the loop it will never be executed.

Additionally, as the _end-directive_ is optional for _loop_ directives like **taskloop**, the `parallel master taskloop` ends with `end do` and GCC 12 rightly complains 4 lines later:
```
   46 |     !$omp end parallel master taskloop
      |                                      1
Error: Unexpected !$OMP END PARALLEL MASTER TASKLOOP statement at (1)
```

Reply:
Fixed; thanks.